### PR TITLE
Cleanup composer installation

### PR DESCRIPTION
**Pull request by evertiro**
Cleans up the composer installation #2437 

## Checks

* [ ] I've updated the changelog.
* [x] I've tested this PR
* [x] This PR is for the `develop` branch not the `stable` branch.
* [x] This PR is complete and ready for review.

### CODE REVIEW

**Review by tomjn — DISMISSED** [review]

**Comment by evertiro** [comment]
@tomjn I think this should also include the provisioner fix. Even though the provisioner is only failing on 20.x, it _is_ a composer issue.

File: .github/workflows/vvv-provisioning.yml [hunk 1]
```diff
@@ -30,13 +30,18 @@ jobs:
           sudo apt-get -q --autoremove --purge remove php*
           sudo apt-get -q autoclean
 
+          # remove pre-installed composer
+          if [ -f /usr/bin/composer ]; then
+            sudo rm -f /usr/bin/composer
```
Comment (msaggiorato) [inline .github/workflows/vvv-provisioning.yml:35]: This doesn't fully install the previously installed composer. Just removes it's executable. Shouldn't we also remove it's config and such?

**Review by msaggiorato — CHANGES_REQUESTED** [review]

File: .github/workflows/vvv-provisioning.yml (hunk 1 above)
Comment (evertiro) [reply .github/workflows/vvv-provisioning.yml:35]: No. I discussed this in a bit more on [the 20.04 issue](https://github.com/Varying-Vagrant-Vagrants/VVV/pull/2389#issuecomment-815518165), but this only runs in the GH action. We _have_ to remove the bin in some environments (20.x+ actions include that file, which breaks our installer), but our composer doesn't have issues with any other files from the base install (honestly, I don't think there really are any), so the less we have to run the faster actions will complete.

Comment (Mte90) [reply .github/workflows/vvv-provisioning.yml:35]: If it is just for the GH action I don't think that is a problem.

File: provision/core/composer/provision.sh [hunk 2]
```diff
@@ -44,17 +42,19 @@ function composer_setup() {
...
     COMPOSER_HOME=/usr/local/src/composer noroot composer --no-ansi global config bin-dir /usr/local/bin
     COMPOSER_HOME=/usr/local/src/composer noroot composer --no-ansi self-update --2 --stable --no-progress --no-interaction
     vvv_info " * Making sure the PHPUnit 7.5 package is available..."
     COMPOSER_HOME=/usr/local/src/composer noroot composer --no-ansi global require --prefer-dist --no-update --no-progress --no-interaction phpunit/phpunit:^7.5
   fi
 
-  vvv_info " * Updating global composer packages..."
-  COMPOSER_HOME=/usr/local/src/composer noroot composer --no-ansi global update --no-progress --no-interaction
-  vvv_success " * Global composer package update completed"
+  if [ -f /usr/local/src/composer/composer.json ]; then
```
Comment (tomjn) [inline provision/core/composer/provision.sh:53]: On MacOS this folder doesn't exist, it's stored elsewhere. We need to figure out how to do this using composer commands

Comment (tomjn) [reply provision/core/composer/provision.sh:53]: I believe `composer config -g` can be used

Comment (tomjn) [reply provision/core/composer/provision.sh:53]: perhaps this:

```sh
~/dev/vvv ᚴ:develop 
❯ composer config -g data-dir
/Users/tarendai/.composer
```

or the `home` config value

Comment (tomjn) [reply provision/core/composer/provision.sh:53]: there's also `composer global show`

Comment (Mte90) [reply provision/core/composer/provision.sh:53]: I can confirm that on debian:
```$ composer config -g data-dir
/home/mte90/.composer
```

Comment (evertiro) [reply provision/core/composer/provision.sh:53]: This is a non-issue. Even on Mac, the VVV environment itself is Linux, and that path is valid.

**Pull request closed without merging**